packages/arb-token-bridge-ui/src/hooks/useNetworks.ts

```diff
@@ -0,0 +1,212 @@
+  sourceChainId,
+  destinationChainId
+}: {
+  sourceChainId: ChainId | undefined
+  destinationChainId: ChainId | undefined
+}): {
+  sourceChainId: ChainId
+  destinationChainId: ChainId
+} {
+  // when both `sourceChain` and `destinationChain` are undefined, default to Ethereum and Arbitrum One
+  if (
+    typeof sourceChainId === 'undefined' &&
+    typeof destinationChainId === 'undefined'
+  ) {
+    return {
+      sourceChainId: ChainId.Ethereum,
+      destinationChainId: ChainId.ArbitrumOne
+    }
+  }
+
+  // only `destinationChainId` is defined
+  if (
+    typeof sourceChainId === 'undefined' &&
+    typeof destinationChainId !== 'undefined'
+  ) {
+    const [defaultSourceChainId] =
+      getPartnerChainsQueryParams(destinationChainId)
+    return { sourceChainId: defaultSourceChainId!, destinationChainId }
+  }
+
+  // only `sourceChainId` is defined
+  if (
+    typeof sourceChainId !== 'undefined' &&
+    typeof destinationChainId === 'undefined'
+  ) {
+    const [defaultDestinationChainId] =
+      getPartnerChainsQueryParams(sourceChainId)
+    return {
+      sourceChainId: sourceChainId,
+      destinationChainId: defaultDestinationChainId!
+    }
+  }
+
+  if (
+    !getPartnerChainsQueryParams(sourceChainId!).includes(destinationChainId!)
+  ) {
+    const [defaultDestinationChainId] = getPartnerChainsQueryParams(
+      sourceChainId!
+    )
+    return {
+      sourceChainId: sourceChainId!,
+      destinationChainId: defaultDestinationChainId!
+    }
+  }
+
+  return {
+    sourceChainId: sourceChainId!,
+    destinationChainId: destinationChainId!
+  }
+}
+
+export type UseNetworksState = {
+  sourceChain: Chain
+  sourceChainProvider: StaticJsonRpcProvider
+  destinationChain: Chain
+  destinationChainProvider: StaticJsonRpcProvider
+}
+
+export type UseNetworksSetStateParams =
+  | {
+      sourceChain: ChainId
+      destinationChain?: ChainId
+    }
+  | {
+      sourceChain?: ChainId
+      destinationChain: ChainId
+    }
+export type UseNetworksSetState = (params: UseNetworksSetStateParams) => void
+
+export function useNetworks(): [UseNetworksState, UseNetworksSetState] {
+  const [
+    { sourceChain: sourceChainId, destinationChain: destinationChainId },
+    setQueryParams
+  ] = useArbQueryParams()
+  const {
+    sourceChainId: validSourceChainId,
+    destinationChainId: validDestinationChainId
+  } = sanitizeQueryParams({
+    sourceChainId,
+    destinationChainId
+  })
+
+  const setState = useCallback(
+    ({ sourceChain, destinationChain }: UseNetworksSetStateParams) => {
+      const {
+        sourceChainId: validSourceChainId,
+        destinationChainId: validDestinationChainId
+      } = sanitizeQueryParams({
+        sourceChainId: sourceChain,
+        destinationChainId: destinationChain
+      })
+      setQueryParams({
+        sourceChain: validSourceChainId,
+        destinationChain: validDestinationChainId
+      })
+    },
+    [setQueryParams]
+  )
```

Comment:
Let's update the params from `sourceChain` and `destinationChain` to `sourceChainId` and `destinationChainId` since it's more accurate

```suggestion
  const setState = useCallback(
    ({ sourceChainId, destinationChainId }: UseNetworksSetStateParams) => {
      const {
        sourceChainId: validSourceChainId,
        destinationChainId: validDestinationChainId
      } = sanitizeQueryParams({
        sourceChainId,
        destinationChainId
      })

      setQueryParams({
        sourceChain: validSourceChainId,
        destinationChain: validDestinationChainId
      })
    },
    [setQueryParams]
  )
```